exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SchemaUtils.kt

```diff
@@ -513,56 +514,97 @@ object SchemaUtils {
      */
     fun checkMappingConsistence(vararg tables: Table, withLogs: Boolean = true): List<String> {
         if (withLogs) {
-            checkExcessiveIndices(tables = tables)
+            checkExcessiveForeignKeyConstraints(tables = tables, withLogs = true)
+            checkExcessiveIndices(tables = tables, withLogs = true)
         }
         return checkMissingIndices(tables = tables, withLogs).flatMap { it.createStatement() }
     }
 
     /**
-     * Checks all [tables] for any that have more than one defined foreign key constraint or index and
-     * logs the findings.
+     * Log Exposed table mappings <-> real database mapping problems and returns DDL Statements to fix them, including
+     * DROP/DELETE statements (unlike [checkMappingConsistence])
+     */
+    private fun mappingConsistenceRequiredStatements(vararg tables: Table, withLogs: Boolean = true): List<String> {
+        return checkMissingIndices(tables = tables, withLogs).flatMap { it.createStatement() } +
+            checkExcessiveForeignKeyConstraints(tables = tables, withLogs).flatMap { it.dropStatement() } +
+            checkExcessiveIndices(tables = tables, withLogs).flatMap { it.dropStatement() }
+    }
+
+    /**
+     * Checks all [tables] for any that have more than one defined index and logs the findings.
      *
-     * If found, this function also logs the SQL statements that can be used to drop these constraints.
+     * If found, this function also logs and returns the SQL statements that can be used to drop these constraints.
      */
-    fun checkExcessiveIndices(vararg tables: Table) {
-        val excessiveConstraints = currentDialect.columnConstraints(*tables).filter { it.value.size > 1 }
+    @Suppress("NestedBlockDepth")
+    private fun checkExcessiveIndices(vararg tables: Table, withLogs: Boolean): List<Index> {
```

Comment:
The function `checkExcessiveIndices` was previously checking both excessive indices and excessive constraints, so I split it.

Reply:
Maybe I'm missing something, but I don't think we should be completely removing a public function from the API. At least not without putting it through a deprecation cycle and informing users of the breaking change with some validation as to why it's being made private. Even if it's probably not a huge impact since it only logged warnings.

Reply:
Agreed. I must have made it private by mistake. It will remain public, but its behaviour has changed and it only deals with indices now, rather than indices and constraints. A statement of a breaking change in the release would suffice, right?

Reply:
Yup, and a reference to the function to use to get info about excessive constraints in case they still need that behavior.